5.0 Draft/IDD/IDDs Service Registry/eu.arrowhead.service-discovery-http-json.yml

```diff
@@ -0,0 +1,631 @@
+          description: A human-readable explanation of why the request was rejected.
+          type: string
+
+    ErrorExpressionIsInvalid:
+      description: |
+        Indicates how a search expression, part of a query or deregistration attempt, is invalid.
+      allOf:
+        - $ref: '#/components/schemas/Error'
+        - type: object
+          properties:
+            expression:
+              description: |
+                Identifies the first invalid expression in the original request, as well as
+                providing an explanation for why it is invalid.
+              type: object
+              properties:
+                offset:
+                  description: |
+                    The position of the offending expression, from the first given in the URI of the
+                    rejected request.
+                  type: number
+                  format: int32
+                  minimum: 0
+                  example: 2
+                message:
+                  description: A human-readable explanation of why this record is invalid.
+                  type: string
+                  example: "Unexpected character '=' at position 20 of 'eu.arrowhead.service=registry'."
+
+    ErrorRecordIsInvalid:
+      description: |
+        Indicates how a record, provided as part of a registration attempt, is invalid.
+      allOf:
+        - $ref: '#/components/schemas/Error'
+        - type: object
+          properties:
+            record:
+              description: |
+                Identifies the first invalid record in the original request, as well as providing
+                an explanation for why it is invalid.
+              type: object
+              properties:
+                offset:
+                  description: |
+                    The position of the offending record, from the first given in the rejected
+                    request.
+                  type: number
+                  format: int32
+                  minimum: 0
+                  example: 4
+                message:
+                  description: A human-readable explanation of why this record is invalid.
+                  type: string
+                  example: "Required property missing: 'interfaces'."
+
+    Metadata:
+      description: |
+        Additional details of relevance when consuming a service. What metadata is made available
+        depends on the type of the service holding the metadata. Possible examples of metadata are
+        HTTP base paths, message size restrictions and caching directives.
+      type: object
+      example: {"basePath":"v2"}
+
+    ProviderIdentifier:
+      description: |
+        An array of bytes identifying a particular service provider. The bytes are expressed using
+        the [URL safe Base 64 encoding](https://datatracker.ietf.org/doc/html/rfc4648#section-5).
+      type: string
+      pattern: ^[A-Za-z0-9-_]*={0,2}$
+      example: MWtzOXIpaUEsOyEhc1NsTwo=
+
+    QueryExpression:
```

Comment:
This is very complex and does not really put any constraints on the expression. Makes it difficult to test and also implement. I'd suggest going with a proper schema in a request body instead.

Reply:
The regular expression can be used by an implementation to add constraints. I agree, however, that a request body in a POST request would be easier to use and implement.

Reply:
If we want the Arrowhead service registry to be somewhat compatible or in line with commercial solutions (Consul) maybe we can take their approach of simpler filter strings? (https://developer.hashicorp.com/consul/api-docs/features/filtering)